src/query/expression/src/new_register.rs

```diff
@@ -0,0 +1,637 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code is generated by src/query/codegen/src/writes/register.rs. DO NOT EDIT.
+
+use std::ops::BitAnd;
+
+use crate::property::Domain;
+use crate::types::nullable::NullableColumn;
+use crate::types::nullable::NullableDomain;
+use crate::types::*;
+use crate::values::Value;
+use crate::EvalContext;
+use crate::Function;
+use crate::FunctionContext;
+use crate::FunctionDomain;
+use crate::FunctionEval;
+use crate::FunctionRegistry;
+use crate::FunctionSignature;
+
+pub fn vectorize_1_arg<I1: ArgType, O: ArgType>(
+    func: impl Fn(I1::ScalarRef<'_>, &mut EvalContext) -> O::Scalar + Copy + Send + Sync,
+) -> impl Fn(Value<I1>, &mut EvalContext) -> Value<O> + Copy + Send + Sync {
+    move |arg1, ctx| {
+        let generics = ctx.generics.to_vec();
+        let iter = (0..ctx.num_rows).map(|index| {
+            let arg1 = unsafe { arg1.index_unchecked(index) };
+            func(arg1, ctx)
+        });
+        let col = O::column_from_iter(iter, &generics);
+        Value::Column(col)
+    }
+}
+
+pub fn vectorize_2_arg<I1: ArgType, I2: ArgType, O: ArgType>(
+    func: impl Fn(I1::ScalarRef<'_>, I2::ScalarRef<'_>, &mut EvalContext) -> O::Scalar
+    + Copy
+    + Send
+    + Sync,
+) -> impl Fn(Value<I1>, Value<I2>, &mut EvalContext) -> Value<O> + Copy + Send + Sync {
+    move |arg1, arg2, ctx| {
+        let generics = ctx.generics.to_vec();
+        let iter = (0..ctx.num_rows).map(|index| {
+            let arg1 = unsafe { arg1.index_unchecked(index) };
```

Comment:
there will be match case inside the loop, did you check the performance benches?

Reply:
```
Gnuplot not found, using plotters backend
bench_kernels/register_old/10240
                        time:   [6.9478 µs 6.9775 µs 7.0112 µs]
Found 10 outliers among 100 measurements (10.00%)
  10 (10.00%) high severe
bench_kernels/register_new/10240
                        time:   [3.1606 µs 3.1758 µs 3.1931 µs]
Found 10 outliers among 100 measurements (10.00%)
  10 (10.00%) high severe
bench_kernels/register_old/102400
                        time:   [67.850 µs 68.373 µs 68.960 µs]
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) high mild
  3 (3.00%) high severe
bench_kernels/register_new/102400
                        time:   [32.901 µs 33.321 µs 33.816 µs]
Found 15 outliers among 100 measurements (15.00%)
```

https://logpaste.com/QjggJAAN

Reply:
I think it's ok to keep old_register for  `1_arg` and `2_arg`.

For multi args we can use new to reduce code expand

Reply:
Seems like the new one is even faster?

Reply:
It's revered typo